### ISSUE TITLE
Vue SFC Rollup Plugin attempt

### ISSUE THREAD

**Issue by Elliotclyde** [preactjs/wmr#382]
Hi everybody. 

Great work on wmr. It's so powerful and awesome for Preact prototypes - complete gamechanger. :+1:

Unfortunately I couldn't get vue SFCs working with the rollup plugin. 

Could anyone advise whether this is maybe impossible to do with wmr at this stage? Or perhaps am I missing some configuration here:

Here is my attempt and the steps I've taken:

`npm install -D wmr
`

Added the following to the package.json

`
"type":"module" 
`

and 

`
 "scripts": {
		"start": "wmr",
		"build": "wmr build",
		"serve": "wmr serve --http2"
	},
`

npm install rollup-plugin-vue
npm install @vue/compiler-sf
touch wmr.config.js

then filled with: 

```
//wmr.config.js
import vuePlugin from 'rollup-plugin-vue';

export default async function({ plugins }) {
    plugins.push(
        vuePlugin({
        target:"browser"
      })
    );
  }

```

mkdir public
touch public/index.html

Filled with:

```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    <div id="app"></div>
    <script type="module" src="./index.js"></script>
</body>
</html>

```
touch public/index.js

Filled with:

```
// public/index.js
import Vue from 'vue'
import App from './app.vue'

new Vue({
  el: '#app',
  template: '<App/>',
  components: { App }
})
```

And finally:

touch public/app.vue

Filled with:

```
// public/app.vue
<template>
  <p>{{ greeting }} World!</p>
</template>

<script>
module.exports = {
  data: function() {
    return {
      greeting: "Hi there"
    };
  }
};
</script>

<style scoped>
p {
  font-size: 2em;
  text-align: center;
}
</style> 
```

At the moment I'm getting a Mime type error in the browser: 

"Failed to load module script: The server responded with a non-JavaScript MIME type of "". Strict MIME type checking is enforced for module scripts per HTML spec"

And running build I get the following error: 

"The "fileName" or "name" properties of emitted files must be strings that are neither absolute nor relative paths and do not contain invalid characters, received "app.vue?vue&type=style&index=0&id=44bfdfee&scoped=true&lang.css"."

Thanks so much for your help: :)

**Comment by Elliotclyde** [comment]
And I'm insinuating I'm the incompetent one here haha. Not the Preact team

**Comment by JoviDeCroock** [comment]
Hey,

I would leave the incompetent stuff behind as we are in uncharted territory here 😅 my initial thought would be that rollup plugin vue will only handle building and as we are tackling an additional aspect in wmr (serving) we'll most likely need something more.

I headed over to vite to confirm this assumption and this seems to be the case as they seem to use a custom plugin as well.

**Comment by Elliotclyde** [comment]
Oh yup. To be clear i was 100% talking about ***myself*** being incompetent at configuring the tools😂 but yeah it's uncharted territory so we're all in the same boat, as you said.

And to confirm: the vue rollup plugin is only built to handle the build stage. So if you did want to use the wmr dev server with vue components you'd need to write a custom plugin to handle this.

I understand that the happy path would be to use vite but i was curious to see if this was posaible w wme

**Comment by JoviDeCroock** [comment]
It's a good point, we should aim at having a stable set of starters for all these frameworks. Personally I'm less experienced with the setup behind Vue, I'd gladly help out but I'd have to get acquainted with how it works, I think the templates we'll first get out are the ones closer to Preact, like React, ...

**Comment by Elliotclyde** [comment]
I managed to get Vue SFC's working in the build step of wmr! 

I created a simplified version of the rollup vue plugin. [Here's the repo](https://github.com/Elliotclyde/wmr-vue-plugin). 

It seems like the build-step issue was something to do with the way the vue rollup plugin uses query parameters which confused wmr. More work to be done to get these going in the dev server! But I'll close the issue. 

Thanks for your help Jovi!
